### PR TITLE
docs(cta): light DOM CSS 

### PR DESCRIPTION
**Pull request by nikkimk**
## What I did

1. Attempted to replicate as much of CTA styles as light DOM CSS using the `:defined` pseudo-class as a demo.
2. Documented the light DOM CSS on the CTA Code docs page.

## Testing Instructions

1. Check the [CTA Code docs, Custom Light DOM CSS](https://deploy-preview-1183--red-hat-design-system.netlify.app/elements/call-to-action/code/#custom-light-dom-css).
2. Check the [proof-of-concept demo](https://deploy-preview-1183--red-hat-design-system.netlify.app/elements/call-to-action/demo/no-javascript-customization).
4. Check the [light DOM CSS file](https://github.com/RedHat-UX/red-hat-design-system/pull/1183/files#diff-b20fff004276871932356783cc8a2c619c817236ef490264484de5b4b3ad5bb1).

## Notes to Reviewers

I'm open to suggestions on how better to maintain and document this.

### CODE REVIEW

**Comment by zhawkins** [comment]
@nikkimk Allen verified this works as expected. 🎉 🥳 🎈

**Review by bennypowers — APPROVED** [review]
I think this approach balances practicality and maintainability

**Comment by markcaron** [comment]
Based on a past convo, we need to package this lightdom CSS file as a physical artifact within RH CTA.

What can we do to revisit this and get this out ASAP in Chansey?

File: elements/rh-cta/demo/nojs.css [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  display: inline !important;
+  border: none !important;
+  background-color: transparent !important;
+  text-decoration: var(--rh-cta-text-decoration);
+  color: inherit !important;
+  font-family: inherit !important;
+  font-size: inherit !important;
+  font-weight: inherit !important;
+  line-height: inherit !important;
+  padding: var(--rh-space-lg, 16px) 0 !important;
```
[the remark this answers is not in the document]
Comment (markcaron) [reply elements/rh-cta/demo/nojs.css:183]: @nikkimk ^

**Review by zeroedin — COMMENTED** [review]
Deploy preview seems broken.

https://deploy-preview-1183--red-hat-design-system.netlify.app/elements/call-to-action/demo/no-javascript-customization/

**Comment by nikkimk** [comment]
> Deploy preview seems broken.
> 
> https://deploy-preview-1183--red-hat-design-system.netlify.app/elements/call-to-action/demo/no-javascript-customization/

@bennypowers this is the case in DP for both cta and subnav.  What am I missing?

**Comment by zeroedin** [comment]
> > Deploy preview seems broken.
> > https://deploy-preview-1183--red-hat-design-system.netlify.app/elements/call-to-action/demo/no-javascript-customization/
> 
> @bennypowers this is the case in DP for both cta and subnav. What am I missing?

The issue lies in handling of the aliases, these are unique to `rh-subnav: Subnavigation` `rh-cta: Call to Action` and Primary/Secondary navigation.  Secondary nav works as its alias is `Navigation (secondary)` and special characters are ignored in the slug creation so it still generates `navigation-secondary`.  Where as cta is trying to generate `rh-call-to-action` for the lightdom css url.

**Review by zeroedin — COMMENTED** [review]
@bennypowers PTAL at the code I pushed here that reworks the lightdom css url rewriting when aliases are involved.  Feel free to improve upon it if you deem necessary.

File: docs/_plugins/rhds.cjs [hunk 2]
```diff
@@ -153,7 +132,42 @@ module.exports = function(eleventyConfig, { tagsToAlphabetize }) {
...
+    const { aliases } = pfeconfig;
+
+    if (inputPath === './docs/elements/demos.html' ) {
+      const tagNameMatch = outputPath.match(/\/elements\/(?<tagName>[-\w]+)\/demo\//);
+      if (tagNameMatch) {
+        const { tagName } = tagNameMatch.groups;
+
+        // slugify the value of each key in aliases creating a new cloned copy
+        const modifiedAliases = { ...aliases };
+        for (const key in aliases) {
```
Comment (bennypowers) [inline docs/_plugins/rhds.cjs:147]: syntax nit, for these kinds of things, `fromEntries` can be preferred, as it's terser, doesn't require the hasOwnProperty call, and focuses the reader's attention on the transformation in the mapping function. it also returns a new object, which let's us skip the spread in the previous line and encourages us not to mutate the inputs, which cuts down on mental execution

```js
const slugifiedAliases = Object.fromEntries(Object.entries(aliases).map(([key, value]) =>
  [slugify(key, { strict: true, lower: true }), value]));
```

**Review by bennypowers — COMMENTED** [review]
notes on the transform.

My earlier approval was for the documentation approach. It's still not correct that cta requires lightdom css, but i won't block this from merging.

**Pull request merged**